proto-lens/src/Data/ProtoLens/TextFormat.hs

```diff
@@ -203,37 +236,55 @@ modifyField (RepeatedField _ f) value = over f (value :)
-makeValue Fixed64Field (Parser.IntValue x) = Right (fromInteger x)
-makeValue SFixed32Field (Parser.IntValue x) = Right (fromInteger x)
-makeValue SFixed64Field (Parser.IntValue x) = Right (fromInteger x)
-makeValue FloatField (Parser.IntValue x) = Right (fromInteger x)
-makeValue DoubleField (Parser.IntValue x) = Right (fromInteger x)
-makeValue BoolField (Parser.IntValue x)
+makeValue :: forall value. Registry -> FieldTypeDescriptor value -> Parser.Value -> Either String value
+makeValue _ Int32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ Int64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ UInt32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ UInt64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SInt32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SInt64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ Fixed32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ Fixed64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SFixed32Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ SFixed64Field (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ FloatField (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ DoubleField (Parser.IntValue x) = Right (fromInteger x)
+makeValue _ BoolField (Parser.IntValue x)
     | x == 0 = Right False
     | x == 1 = Right True
     | otherwise = Left $ "Unrecognized bool value " ++ show x
-makeValue DoubleField (Parser.DoubleValue x) = Right x
-makeValue FloatField (Parser.DoubleValue x) = Right (realToFrac x)
-makeValue BoolField (Parser.EnumValue x)
+makeValue _ DoubleField (Parser.DoubleValue x) = Right x
+makeValue _ FloatField (Parser.DoubleValue x) = Right (realToFrac x)
+makeValue _ BoolField (Parser.EnumValue x)
     | x == "true" = Right True
     | x == "false" = Right False
     | otherwise = Left $ "Unrecognized bool value " ++ show x
-makeValue StringField (Parser.ByteStringValue x) = Right (Text.decodeUtf8 x)
-makeValue BytesField (Parser.ByteStringValue x) = Right x
-makeValue EnumField (Parser.IntValue x) =
+makeValue _ StringField (Parser.ByteStringValue x) = Right (Text.decodeUtf8 x)
+makeValue _ BytesField (Parser.ByteStringValue x) = Right x
+makeValue _ EnumField (Parser.IntValue x) =
     maybe (Left $ "Unrecognized enum value " ++ show x) Right
         (maybeToEnum $ fromInteger x)
-makeValue EnumField (Parser.EnumValue x) =
+makeValue _ EnumField (Parser.EnumValue x) =
     maybe (Left $ "Unrecognized enum value " ++ show x) Right
         (readEnum x)
-makeValue MessageField (Parser.MessageValue x) = buildMessage x
-makeValue GroupField (Parser.MessageValue x) = buildMessage x
-makeValue f val = Left $ "Type mismatch parsing text format: " ++ show (f, val)
+makeValue reg field@MessageField val@(Parser.MessageValue typeUri x)
+    | Just AnyMessageDescriptor { anyTypeUrlLens, anyValueLens } <- matchAnyMessage field =
+        case typeUri of
```

Comment:
Can we move these pattern matches into to the guard?  If decoding the Any fails, we should just default to displaying the raw typeUrl and value (i.e. l.282) rather than failing the decode.

Also, maybe add a test for that edge case, if it's straightforward.